### PR TITLE
RA-921

### PR DESCRIPTION
**Pull request by jdegraft**
IdGen upgraded to work with Hibernate 4.0 (OpenMRS Platform 2.0)

### CODE REVIEW

**Comment by mogoodrich** [comment]
@mseaton lmk if you are going to take care of this one, or if you'd like me to.

File: pom.xml [hunk 1]
```diff
@@ -84,7 +84,7 @@
 	</dependencies>
 
 	<properties>
-		<openMRSVersion>1.9.2</openMRSVersion>
+		<openMRSVersion>1.10.2</openMRSVersion>
```
Comment (mseaton) [inline pom.xml]: Any reason for 1.10.2 vs. 1.9.9?

File: omod/src/main/resources/config.xml [hunk 2]
```diff
@@ -11,10 +11,10 @@
 		See <a href="http://openmrs.org/wiki/Idgen_Module">http://openmrs.org/wiki/Idgen_Module</a> for more information
 	</description>
 	<activator>@MODULE_PACKAGE@.IdgenModuleActivator</activator>
-	<require_version>1.9.2</require_version>
 	<require_modules>
 		<require_module>org.openmrs.module.webservices.rest</require_module>
 	</require_modules>
+	<require_version>1.11.3, 1.10.2 - 1.10.*, 1.9.9 - 1.9.*</require_version>
```
Comment (mseaton) [inline omod/src/main/resources/config.xml]: Curious about this?  Do we need those upper bounds or are they implied?  And if we need them, don't we also need them on the 1.11 line and above?

File: pom.xml (hunk 1 above)
Comment (jdegraft) [inline pom.xml]: Those were the versions I understand was right. It is a bit contradictory to the requires statement but these changes followed a template given in TRUNK-4713.

File: omod/src/main/resources/config.xml (hunk 2 above)
Comment (jdegraft) [inline omod/src/main/resources/config.xml]: These changes were according to the template given in TRUNK-4713. There was a brief talk discussion and the template was the consensus approach.

Comment (mseaton) [inline omod/src/main/resources/config.xml]: @dkayiwa @wluyima @rkorytkowski  - just curious as to your thoughts on this.  Are we recommending the above and is it necessary?  Would just having 1.9.9,1.10.2,1.11.3 be sufficient, without the wildcard ranges, or are they needed?

Comment (mseaton) [inline omod/src/main/resources/config.xml]: Ok never mind I see why they are needed and that makes sense...

File: pom.xml (hunk 1 above)
Comment (mseaton) [inline pom.xml]: It's fine.  You can only choose one version.  I typically choose the lowest supported version.  But an equally (or more) valid approach could be the most widely used version.

**Pull request merged**